webapp/app/js/directives/clonejobs.js

```diff
@@ -1069,6 +1175,84 @@ treeherder.directive('thCloneJobs', [
             tableEl.append(row);
         }
     };
+    
+    var traceParent = function(el, level){
+
+        var i;
+        for (i=0; i<level; i++) {
+            el = el.parent();
+        }
+
+        return el;
+    }
+
+    var traceJob = function(elem, step, clss){
+
+        var elemsList, jobFound, jobElem, reverseList, tempList, jobsList;
+
+        elemsList = [elem];
+        jobFound = false;
+        jobElem = null;
+        reverseList = false;
+
+        while (!jobFound && elemsList.length>0) {
+
+            tempList = [];
+
+            angular.forEach(elemsList, function(tempElem) {
+
+                if (!jobFound) {
+
+                    if (tempElem.hasClass(clss) && tempElem.css('display')!='none') {
+
+                        jobFound = true;
+                        jobElem = $(tempElem);
+
+                    } else {
+
+                        if (tempElem.children().length>0) {
+
+                            angular.forEach(tempElem.children(), function(tempChild) {
+
+                                if ($(tempChild).hasClass('platform-group')) {
+
+                                    jobsList = $($(tempChild).children()[1]).children();
+                                    for (var i=0; i<jobsList.length; i++) {
+                                        tempList.push($(jobsList[i]));
+                                    }
+                                    
+                                    reverseList = true;
+
+                                } else if ($(tempChild).hasClass('job-btn')) {
+
+                                    tempList.push($(tempChild));
+                                    reverseList = true;
+
+                                } else {
+                                    tempList.push($(tempChild));
+                                }
+                                
+                            })
+                        }
+                    }
+                }
+            })
```

Comment:
If I am reading the block correctly, this is missing a trailing semicolon above. If you run the files through [jshint](http://jshint.com/) it will expose these types of things.